File: sorting-algos/java/insertion.java
class Main
{
      static void insertionSort(int arr[]){         
          for(int i = 0 ; i < arr.length ; i++){
              for(int j = i ; j > 0 ; j--){
                  if(arr[j] < arr[j - 1]){
                      int temp = arr[j-1];                  
                      arr[ j -1 ] = arr[j];
                      arr[j] = temp;
                  }
              }
          }
          
         for(int i = 0 ; i < arr.length ; i++){
             System.out.println(arr[i]);
          }
    }
    
	public static void main(String[] args) {
	    int arr[] = {2,5,7,8,4,6,3};
	    insertionSort(arr);
		
	}
}

File: sorting-algos/ts/insertion.js
// [7, 5, 2, 8, 4, 6, 3] 
// i = 3, j = 0
function insertionSort(arr) {
    var i = 1, j, temp;
    while (i < arr.length) {
        j = i;
        while (j > 0 && arr[j - 1] < arr[j]) {
            temp = arr[j];
            arr[j] = arr[j - 1];
            arr[j - 1] = temp;
            j = j - 1;
        }
        i++;
    }
    return arr;
}
var arr = [2, 5, 7, 8, 4, 6, 3];
console.log(insertionSort(arr));
